api/desecapi/views.py

```diff
@@ -133,16 +135,15 @@ def perform_create(self, serializer):
 
         try:
             obj = serializer.save(owner=self.request.user)
-        except Exception as e:
-            if str(e).endswith(' already exists'):
-                ex = ValidationError(detail={
-                    "detail": "This domain name is unavailable.",
-                    "code": "domain-unavailable"}
-                )
-                ex.status_code = status.HTTP_409_CONFLICT
-                raise ex
-            else:
+        except (IntegrityError, PdnsException) as e:
+            if isinstance(e, PdnsException) and not str(e).endswith(' already exists'):
                 raise e
+            ex = ValidationError(detail={
+                "detail": "This domain name is unavailable.",
+                "code": "domain-unavailable"}
+            )
+            ex.status_code = status.HTTP_409_CONFLICT
+            raise ex
```

Comment:
Just as a general note: While refactoring for #97, I found that there is at least one more type how things can go wrong, which is a Django `ValidationError`. I hence believe there are in total four ways that need to be considered when creating objects (most notably Domains and RRSets):

1. IntegrityError coming from the api's database
2. PdnsException coming from, well, powerdns
3. Django's ValidationError coming from the model layer
4. DRF's ValidationError coming from the serializer

I'm not saying all of them are needed o be caught here, but just wanted to note this for future reference.